cmd/api-server/cli/root_test.go

```diff
@@ -0,0 +1,70 @@
+package cli
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/docker/compose-on-kubernetes/internal/keys"
+
+	"github.com/stretchr/testify/assert"
+	genericoptions "k8s.io/apiserver/pkg/server/options"
+	certutil "k8s.io/client-go/util/cert"
+)
+
+func TestGenerateBundleAndMerge(t *testing.T) {
+	hostname, err := os.Hostname()
+	assert.NoError(t, err)
+	oldOpts := &apiServerOptions{
+		RecommendedOptions: &genericoptions.RecommendedOptions{
+			SecureServing: &genericoptions.SecureServingOptionsWithLoopback{
+				SecureServingOptions: &genericoptions.SecureServingOptions{},
+			},
+		},
+		serviceName:      "old-service-name",
+		serviceNamespace: "old-namespace",
+	}
+	err = generateCertificateIfRequired(oldOpts)
+	if oldOpts.caBundleFile != "" {
+		defer os.RemoveAll(filepath.Dir(oldOpts.caBundleFile))
+	}
+	assert.NoError(t, err)
+	oldCA, err := certutil.CertsFromFile(oldOpts.caBundleFile)
+	assert.NoError(t, err)
+	assert.Len(t, oldCA, 1)
+	assert.Equal(t, "compose-api-ca-"+strings.ToLower(hostname), oldCA[0].Subject.CommonName)
```

Comment:
Can you just add empty lines and/or comments between logical blocks? It will be easier to review/maintain.